graylog2-server/src/test/java/org/graylog/plugins/pipelineprocessor/rulebuilder/parser/ConditionParserTest.java

```diff
@@ -151,12 +150,12 @@ public void generate_WhenRuleConditionsContainsMultipleFunctions() {
                         .build(),
                 RuleBuilderStep.builder().function(FUNCTION2_NAME).negate().build()
         );
-        assertThat(conditionParser.generate(steps)).isEqualTo("""
-                  true
-                  && function1(
+        assertThat(conditionParser.generate(steps, RuleBuilderStep.Operator.AND, 1)).isEqualTo("""
+                  function1(
                     required : "val1"
                   )
-                  && ! function2()
+                    AND
+                  ! function2()
                 """.stripTrailing());
     }
 
```

Comment:
Add a TODO comment to `generate_WhenRuleConditionsContainsOneFragment` or delete it.